libs/zzz/page-optimize-pando/src/OptSelector.tsx

```diff
@@ -1,19 +1,32 @@
 import type { CharOpt, ICachedCharacter } from '@genshin-optimizer/zzz/db'
 import { useDatabaseContext } from '@genshin-optimizer/zzz/db-ui'
+import { getFormula } from '@genshin-optimizer/zzz/formula'
 import { OptimizationTargetSelector } from '@genshin-optimizer/zzz/formula-ui'
+import { useMemo } from 'react'
 
 export function OptSelector({
   character: { key: characterKey },
-  charOpt: { targetName, targetSheet },
+  charOpt: { targetName, targetSheet, targetDamageType1, targetDamageType2 },
 }: {
   charOpt: CharOpt
   character: ICachedCharacter
 }) {
   const { database } = useDatabaseContext()
+
+  const formula = getFormula(targetSheet, targetName)
+  const tag = useMemo(
+    () =>
+      formula?.tag && {
```

Comment:
this construction is kinda cringy being repeated anywhere. idk if it warrants refactoring a bit